src/components/AddNicknameForm/AddNicknameForm.js

```diff
@@ -41,12 +44,40 @@ class AddNicknameForm extends React.Component {
     const {
       nickname,
     } = this.state
-
+    const httpErrors = {
+      conflict: 409,
+      unprocessableEntity: 422,
+    }
     event.preventDefault()
 
     this.setState({ submitting: true })
 
-    await addNickname(user, { attributes: { nick: nickname } })
+    const response = await addNickname(user, { attributes: { nick: nickname } })
+
+    if (isError(response)) {
+      const { meta, payload } = response
+      let errorMessage = 'Unknown error occured.'
+
+      if (HttpStatus.isClientError(meta.response.status)) {
+        errorMessage = payload.errors && payload.errors.length ? payload.errors[0].title : 'Client communication error'
+      }
+
+      if (HttpStatus.isServerError(meta.response.status)) {
+        errorMessage = 'Server communication error'
+      }
+
+      if (meta.response.status === httpErrors.conflict) {
```

Comment:
HttpStatus has all of these defined, no need to define them again in the object above.

https://github.com/FuelRats/fuelrats.com-utils/blob/master/src/http/index.js#L120